specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/utils.ts

```diff
@@ -176,26 +181,37 @@ export function matchSelectedFiles(
 
 export function resolveFileNames(
   previousFile: UnBoundFile,
-  getFormatted: (
-    rawName: number | string | null | undefined
-  ) => string | undefined,
+  getFormatted: (rawName: number | string | undefined) => string | undefined,
   formatter?: UiFormatter
 ): PartialUploadableFileSpec {
-  let nameToParse: string | undefined;
+  const splitName = stripFileExtension(previousFile.name);
+  let nameToParse = splitName;
   if (
-    formatter === undefined ||
-    formatter.fields.some((field) => field instanceof formatterTypeMapper.regex)
-  )
-    nameToParse = stripFileExtension(previousFile.name);
-  else {
+    formatter?.fields.every(
+      (field) => !(field instanceof formatterTypeMapper.regex)
+    ) === true
+  ) {
     const formattedLength = formatter.fields.reduce(
       (length, field) => length + field.size,
       0
     );
     nameToParse = previousFile.name.slice(0, formattedLength);
   }
   previousFile.parsedName = f.maybe(nameToParse, getFormatted);
-
+  /*
+   *  If there is one numeric field, but naive limiting of length
+   *  didn't work, split by extension to try matching it.
+   *  Done to catch cases like 2.txt, and to treat it as 000000002.
+   *  Allows treating 2 as ABC-DEF-2.
+   *  Cannot handle more than one numeric field because number is ambiguous
+   */
+  if (
+    formatter?.fields.filter(
+      (field) => field instanceof formatterTypeMapper.numeric
+    ).length === 1 &&
+    previousFile.parsedName === undefined
+  )
+    previousFile.parsedName = getFormatted(splitName);
```

Comment:
seems like a hacky solution.

can you instead apply the (catalog number) field formatter on the file name?
(i.e in the form if you type "2", it would save "00000002" if you have a numeric field formatter configured)

that would handle all the cases, not just the single numeric field case

Reply:
that is what getFormatted does (it comes from the field formatter). So, if this is catalog number, getFormatted will automatically be the catalog number formatter. Maybe I could just do `previousFile.parsedName = getFormatted(splitName)` instead of restricting it to just numeric. So, it would be
`if (previousFile.parsedName === undefined) previousFile.parsedName = getFormatted(splitName)`.